docs/docs-beta/docs/guides/build/components/setting-up-a-deployment.md

```diff
@@ -0,0 +1,177 @@
+│   └── code-location-1
+│       ├── code_location_1
+│       │   ├── __init__.py
+│       │   ├── components
+│       │   ├── definitions.py
+│       │   └── lib
+│       │       ├── __init__.py
+│       ├── code_location_1_tests
+│       │   └── __init__.py
+│       ├── pyproject.toml
+│       └── uv.lock
+└── pyproject.toml
+```
+
+The `code-location-1` directory contains a `pyproject.toml` file that defines
+it as a code location and component library:
+
+```toml
+[tool.dagster]
+module_name = "code_location_1.definitions"
+project_name = "code_location_1"
+
+[tool.dg]
+is_code_location = true
+is_component_lib = true
+```
+
+Let's enter this directory and search for registered component types:
+
+```bash
+$ cd code_locations/code-location-1 && dg component-type list
+
+dagster_components.definitions
+dagster_components.pipes_subprocess_script_collection
+    Assets that wrap Python scripts executed with Dagster's PipesSubprocessClient.
+```
+
+This is the default set of component types available in every new code
+location. We can add to it by installing `dagster-components[sling]`:
+
+```bash
+$ uv add dagster-components[sling]
+```
+
+And now we have a new available component:
+
+```bash
+$ dg component-type list
+
+dagster_components.definitions
+dagster_components.pipes_subprocess_script_collection
+    Assets that wrap Python scripts executed with Dagster's PipesSubprocessClient.
+dagster_components.sling_replication_collection
+```
+
+As stated above, environments are scoped per code location.  `dg` commands will
+only use the environment of `code-location-1` when we are inside the
+`code-location-1` directory.
+
+Let's create another code location to demonstrate this:
+
+```bash
+$ cd ../.. && dg code-location scaffold code-location-2
+
+Creating a Dagster code location at .../my-deployment/code_locations/code-location-2.
+Scaffolded files for Dagster project in .../my-deployment/code_locations/code-location-2.
+```
+
+Now we have two code locations. We can list them with:
+
+```bash
+$ dg code-location list
+
+code-location-1
+code-location-2
+```
+
+And finally, let's check the available component types in `code-location-2`:
+
+```bash
+$ cd code_locations/code-location-2 && dg component-type list
+
+dagster_components.definitions
+dagster_components.pipes_subprocess_script_collection
+    Assets that wrap Python scripts executed with Dagster's PipesSubprocessClient.
+```
+
+As you can see, we are back to only the default list of component types. This
+is because we are now using the environment of `code-location-2`, in which we
+have not installed `dagster-components[sling]`.
+
+For a final step, let's load up our two code locations with `dagster dev`.
+We'll need a workspace.yaml to do this. Create a new file `workspace.yaml` in
+the `my-deployment` directory:
+
+:::note
+`dg` scaffolding functionality is currently under heavy development. In the
+future we will construct this workspace.yaml file for you automatically in the
```

Comment:
note: I think should use this opportunity to deal a final death blow to `workspace.yaml` and just drive from `pyproject.yaml`

Reply:
Agree, but since we don't have that hashed out yet let's leave this as is for now? (Unless it's already possible to specify multiple code locations in pyproject.toml? I've only seen single code location).

Reply:
Wasn't proposing doing that in this PR